From 5ab68b828e94349a61caee727ad42e1d69286fe2 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 30 Jan 2019 15:25:14 +0200
Subject: [PATCH] [SW-2807] fw_upd: do not zero FLOPS3 in preboot

We are about to extend the non-reset flops commands, so to prepare
do not zero the register.
This is needed to allow the extension of FLOPS3 by additional commands.

Change-Id: I8856010e4870888343f10dea4ee9211ca1efea32
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot.c b/preboot.c
index 238e4c7..e34c0b6 100644
--- a/preboot.c
+++ b/preboot.c
@@ -189,8 +189,7 @@
 			try_new_image = true;
 		} else
 			writel(FLOPS_UPD_NOOP, mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3);
-	} else if (modifiable_image_pending) /* ONESHOT */
-		writel(FLOPS_UPD_NOOP, mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3);
+	}
 
 	if (!try_new_image) {
 		spl_image = "stable";
